ui/src/app/domains/views/DomainsList/DomainListHeaderForm/DomainListHeaderForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { Col, Input, Row } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+import type { SchemaOf } from "yup";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+
+type Props = {
+  closeForm: () => void;
+};
+
+export type CreateDomain = {
+  name: string;
+  isAuthoritative: boolean;
+};
+
+const DomainListHeaderForm = ({ closeForm }: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const errors = useSelector(domainSelectors.errors);
+  const saved = useSelector(domainSelectors.saved);
+  const saving = useSelector(domainSelectors.saving);
+
+  // Pattern that matches a domainname.
+  // XXX 2016-02-24 lamont: This also matches "example.com.",
+  // which is wrong.
+  const domainnamePattern = /^([a-z\d]|[a-z\d][a-z\d-.]*[a-z\d])*$/i;
```

Comment:
When writing out a regex like this it's good practice to have examples of what it matches and what it doesn't to aid readers of the code in the future from having to decode the regex. Especially for those who aren't super familiar with it.

When writing an XXX can you also create a follow-up issue so that we don't forget to go back and fix up the issue in the future.

Reply:
I merely ported the regex (and comment) from the angular app. I have not idea what the requirements are for a domain name

Reply:
Ahh ok ok :)

Reply:
I've added an issue to address it here: https://github.com/canonical-web-and-design/maas-ui/issues/2755